### PR TITLE
WIP: ReceivePoller

### PR DESCRIPTION
**Pull request by John-Colvin**
The easiest way to wait on multiple sockets in nanomsg.

This is very much WIP right now, but I thought it would be good to have it seen as I work.

### CODE REVIEW

File: source/nanomsg/wrap.d [hunk 1]
```diff
@@ -509,6 +491,166 @@ private:
...
+        pds[idx].revents = nn_pollfd.init.revents;
+    }
+}
+
+auto receivePoll(NanoSocket*[] sockets)
+{
+    return ReceivePoller!(Yes.blocking)(sockets);
+}
+
+auto receivePoll(NanoSocket*[] sockets, int timeoutMs)
```
Comment (atilaneves) [inline source/nanomsg/wrap.d]: `Duration timeout` please.

File: source/nanomsg/wrap.d [hunk 2]
```diff
@@ -509,6 +491,166 @@ private:
...
+                                     " errno ", nn_errno, ", error: ", nn_strerror(nn_errno)),
+                                file,
+                                line);
+        }
+    }
+    return value;
+}
+
+
+struct PollResult(Flag!"blocking" blocking = Yes.blocking)
```
Comment (atilaneves) [inline source/nanomsg/wrap.d:515]: This could use some DDoc.

File: source/nanomsg/wrap.d [hunk 3]
```diff
@@ -509,6 +491,166 @@ private:
...
+{
+    private NanoSocket*[] sockets;
+    private nn_pollfd[] pds;
+    private PollResult!blocking front_;
+    static if (!blocking)
+        int timeoutMs;
+    enum empty = false;
+
+    static if (!blocking)
+        this(NanoSocket*[] sockets, int timeoutMs)
```
Comment (atilaneves) [inline source/nanomsg/wrap.d]: `Duration` instead of `int` please.

File: source/nanomsg/wrap.d (hunk 1 above)
Comment (John-Colvin) [reply source/nanomsg/wrap.d]: done

File: source/nanomsg/wrap.d (hunk 3 above)
Comment (John-Colvin) [reply source/nanomsg/wrap.d]: done

**Comment by atilaneves** [comment]
CI is failing.

**Comment by atilaneves** [comment]
Removal of "WIP" and test coverage are the only things stopping me from merging this.

**Comment by John-Colvin** [comment]
I'm not happy with the API yet, so I'm going to take it in-house for further use, testing and improvement.

Doing that relies on https://github.com/kaleidicassociates/nanomsg-wrapper/pull/12

**Pull request closed without merging**